src/mlang/backend_ir/bir.ml

```diff
@@ -251,8 +292,8 @@ let rec remove_empty_conditionals (stmts : stmt list) : stmt list =
          | _ -> stmt :: acc)
        [] stmts)
 
-let rec get_used_variables_ (e : expression Pos.marked) (acc : VariableDict.t) :
-    VariableDict.t =
+let rec get_used_variables_ (e : expression Pos.marked) (acc : VariableSet.t) :
```

Comment:
(Nitpicking) I think we have a fold and a map over expressions, so this piece of code could be refactored.

Reply:
Done